src/Components/Components/src/Binding/FormDataProvider.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Binding;
+
+/// <summary>
+/// Provides information about the state of the form being processed.
+/// </summary>
+public abstract class FormDataProvider
+{
+    /// <summary>
+    /// Gets the name of the form associated with the data in this form.
```

Comment:
Nit: This docstring is a little confusing. Maybe just:

```suggestion
    /// Gets the name associated with the data in this form.
```

?